Car_rental_app/website/views.py

```diff
@@ -302,3 +307,31 @@ def checkout_requests():
 
 
 
+
+
+
+@views.route('/admin/sales', methods=['GET', 'POST'])
+def admin_sales():
+    if request.method == 'POST':
+        month = int(request.form['month'])
+        year = int(request.form['year'])
+        
+        # Calculate start and end dates of the specified month
+        start_date = datetime(year, month, 1)
```

Comment:
you can replace your code by this: 
start_date = datetime(year, month, 1)
end_date = datetime(year, month, calendar.monthrange(year, month)[1])

the calendar.monthrange() will make sure that the end date is accurate and also it can handle leap years and varying days of month automatically. More reliable to calculate in my opinion